google/ggrc-core
# GGRC-5646 Export request did not finish due to errors while exporting all controls snapshots

# Issue description

Export request did not finish due to errors while exporting all controls snapshots.
Possible reason - unused objects not released while exporting large amount of data which leads to process being killed when running out of memory.

# Steps to test the changes

1. Open Export page
2. Select OBJECT TYPE 'Snapshots'
3. Select 'SELECT SNAPSHOT OBJECT TYPE' e.g. Control
4. Click Export button
Before changes: Export failed due to server error. occurred, export was not finished
After Changes: Export success (~190000 Snapshots)

Downloading exported **csv** file will fail because of GAE limit of 32Mb response size. This issue should be addressed in another ticket.

# Solution description

Solution is partially based on [this](https://github.com/google/ggrc-core/pull/7618) Pull Request.

- Release unused objects while collecting data.
- Split query into chunks of 500 Snapshots.
- Drop access_control_map cache.
- Change import_exports content field type to CompressedType.
- Add zlib compression/decompression to CompressedType.
- Add migration.
- Fix tests.

# Sanity checklist




# PR Review checklist
